### PR TITLE
Wordpress admin upload `check`

### PR DESCRIPTION
**Pull request by egypt**
Adds a `check` method to `wp_admin_shell_upload` to verify that credentials work. Note that in a default install of WP from Debian repositories, this is insufficient to achieve code execution because the plugins directory is owned by root, so it only returns `Checkcode::Appears` if login was successful rather than `Checkcode::Vulnerable`.

This will also record credentials for the handful of other modules targeting WordPress that require authentication.

## Verification

List the steps needed to make sure this thing works

- [ ] Set up WordPress
- [ ] `use exploit/unix/webapp/wp_admin_shell_upload`
- [ ] set RHOST, USERNAME, and PASSWORD appropriately
- [ ] `check`
- [ ] **Verify** `creds` displays the values you set for USERNAME and PASSWORD
- [ ] `exploit`
- [ ] **Verify** getting a shell still works

### CODE REVIEW

File: lib/msf/core/exploit/http/wordpress/login.rb [hunk 1]
```diff
@@ -14,18 +14,41 @@ def wordpress_login(user, pass, timeout = 20)
...
-      cookies = res.get_cookies
-      # Check if a valid wordpress cookie is returned
-      return cookies if
-        # current Wordpress
-        cookies =~ /wordpress(?:_sec)?_logged_in_[^=]+=[^;]+;/i ||
-        # Wordpress 2.0
-        cookies =~ /wordpress(?:user|pass)_[^=]+=[^;]+;/i ||
-        # Wordpress 2.5
-        cookies =~ /wordpress_[a-z0-9]+=[^;]+;/i
+    cookies = res.get_cookies
```
Comment (FireFart) [inline lib/msf/core/exploit/http/wordpress/login.rb]: @egypt is it intended to remove the `res.redirect?` check here? Has WP removed this?

**Comment by wchen-r7** [comment]
Hi @egypt, it looks like the rspec for Msf::Exploit::Remote::HTTP::Wordpress::Login needs an update so travis can go green?

Please see https://travis-ci.org/rapid7/metasploit-framework/jobs/225373495

**Comment by wchen-r7** [comment]
Labeled as Delay due to the Travis failure.

File: lib/msf/core/exploit/http/wordpress/login.rb [hunk 2]
```diff
@@ -14,18 +14,41 @@ def wordpress_login(user, pass, timeout = 20)
...
+
+    if cookies && (
+      # current Wordpress (2.6+)
+      cookies =~ /wordpress_(?:sec|logged_in_)[^=]+=[^;]+;/i ||
+      # Wordpress 2.5
+      cookies =~ /wordpress_[a-f0-9]+=[^;]+;/i ||
```
Comment (egypt) [inline lib/msf/core/exploit/http/wordpress/login.rb]: This needs to be `[a-z0-9]` for the spec. In my testing, wordpress always produces hex digits here, but it's fine to be more relaxed.

File: lib/msf/core/exploit/http/wordpress/login.rb (hunk 1 above)
Comment (egypt) [inline lib/msf/core/exploit/http/wordpress/login.rb]: The redirect check didn't seem necessary, but there's really no reason to remove it. I'll put it back.

**Comment by egypt** [comment]
Still delayed, waiting on results of some discussions with @jmartin-r7

File: lib/msf/core/exploit/http/wordpress/login.rb [hunk 3]
```diff
@@ -16,14 +16,39 @@ def wordpress_login(user, pass, timeout = 20)
...
+        private_data: pass,
+        private_type: :password,
+      }.merge(service_data)
+
+      core = create_credential(cdata)
+      login_data = { core: core }.merge(service_data)
+
+      create_credential_login(login_data)
+
+      return cookies
```
Comment (jmartin-r7) [inline lib/msf/core/exploit/http/wordpress/login.rb]: Lets revert file's changes if https://github.com/rapid7/metasploit-framework/pull/8327 is an acceptable alternative.

**Comment by jmartin-r7** [comment]
# Release Note

The WordPress admin shell upload module has been enhanced with new documentation and for support the check method.

**Comment by FireFart** [comment]
@jmartin-r7 travis failed with the merge:
```
--- Checking new and changed module syntax with tools/dev/msftidy.rb ---
modules/exploits/unix/webapp/wp_admin_shell_upload.rb - [WARNING] Explicitly requiring/loading msf/core is not necessary
modules/exploits/unix/webapp/wp_admin_shell_upload.rb - [WARNING] Explicitly using self.class in register_* is not necessary
```

**Comment by bwatters-r7** [comment]
# Release Notes
Add a check method to wp_admin_shell_upload to verify that credentials work.

**Pull request merged**